File: Homework/Ngan_Luu/88-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i=m-1
        j=n-1
        for k in range(len(nums1)-1,-1,-1):
            if j==-1:
                break
            if i==-1:
                nums1[k]=nums2[j]
                j-=1
            else:
                if nums1[i]<=nums2[j]:
                    nums1[k]=nums2[j]
                    j-=1
                else:
                    nums1[k]=nums1[i]
                    i-=1

File: Homework/Hoang_Nguyen/lesson3/88_mergeSortedArray.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function (nums1, m, nums2, n) {
  let i1 = m - 1;
  let i2 = n - 1;
  let j = m + n - 1;
  // insert element from tail to head of nums1
  for (; j >= 0; j--) {
    if (i2 < 0) {
      break;
    }
    if (nums1[i1] > nums2[i2]) {
      nums1[j] = nums1[i1];
      i1--;
    } else {
      nums1[j] = nums2[i2];
      i2--;
    }
  }
  // insert remaining elements into nums1
  for (; i2 >= 0; i2--) {
    nums1[j] = nums2[i2];
    j--;
  }
};
